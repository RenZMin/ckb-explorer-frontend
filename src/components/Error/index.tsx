import PCDataNotFoundImage from '../../assets/pc_data_not_found.png'
import MobileDataNotFoundImage from '../../assets/mobile_data_not_found.png'
import { isMobile } from '../../utils/screen'
import { ErrorPanel } from './styled'

export default () => (
  <ErrorPanel>
    <img alt="data not found" src={isMobile() ? MobileDataNotFoundImage : PCDataNotFoundImage} />
  </ErrorPanel>
)
