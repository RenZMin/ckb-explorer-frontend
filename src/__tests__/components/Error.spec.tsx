import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Error from '../../components/Error'
import { ErrorPanel } from '../../components/Error/styled'

describe('Error Component', () => {
  let component: ReactElement

  beforeAll(() => {
    component = <Error />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find(ErrorPanel).children().length).toBe(1)
  })
})
