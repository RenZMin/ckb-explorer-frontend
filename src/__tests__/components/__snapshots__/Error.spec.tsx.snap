// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component shallow renders 1`] = `
<div
  className="sc-bdnxRM jgEPVI"
>
  <img
    alt="data not found"
    src="pc_data_not_found.png"
  />
</div>
`;
